certbot-apache/certbot_apache/_internal/configurator.py

```diff
@@ -225,6 +238,32 @@ def updated_mod_ssl_conf_digest(self):
         """Full absolute path to digest of updated SSL configuration file."""
         return os.path.join(self.config.config_dir, constants.UPDATED_MOD_SSL_CONF_DIGEST)
 
+    def openssl_version(self):
+        """Lazily retrieve openssl version"""
+        if self._openssl_version:
+            return self._openssl_version
+        # Step 1. Check for LoadModule directive
+        try:
+            ssl_module_location = self.parser.modules['ssl_module']
+        except KeyError:
+            return None
+        if not ssl_module_location:
+            return None
+        # Step 2. Grep in the .so for openssl version
+        try:
+            with open(ssl_module_location) as f:
+                contents = f.read()
+        except IOError as error:
+            logger.debug(str(error), exc_info=True)
+            raise errors.PluginError("Unable to open ssl_module file")
```

Comment:
I personally think we may just want to log a warning and leave session tickets enabled in this case too. If we hit this case on a user's system, raising an exception will break the plugin there until we patch it while logging a warning will print and log a message on every run, but keep the plugin working.

Reply:
My original reasoning here was that if this is an error, there's likely a bigger issue at play that will cause failures so might as well error, but I'm also perfectly fine with changing this to a non-breaking error and letting whatever else might have failed elsewhere fail.